tests/conftest.py

```diff
@@ -388,43 +388,38 @@ def suppress_internal_deprecations():
         yield
 
 
-@pytest.fixture
-def with_daemon():
-    """Starts the daemon process and then makes sure to kill it once the test is done."""
-    import subprocess
-    import sys
+@pytest.fixture(scope='session')
+def daemon_client(aiida_profile):
```

Comment:
Is this fixture actually ever needed by the user or would `with_daemon` and `without_daemon` suffice?

Note that in `without_daemon`, the daemon is currently stopped twice (not a big deal, but just to say that perhaps the logic is not entirely shared between the to `with...` fixtures, and copying a little code between `with_daemon` and `without_daemon` would be ok).

I see that `daemon_client` existed before, so one can also decide to keep it but deprecate it.

Reply:
> Is this fixture actually ever needed by the user or would with_daemon and without_daemon suffice?

It is true that it probably is not necessary to use these in tests themselves, but it does make sense to have. Note that the scope of `daemon_client` is different from `with_daemon` and `without_daemon`. The latter are function scoped whereas the client is session scoped. So anyone we would need a base fixture that creates the client and ensures the daemon stops at the end of the session.

This also explains why `without_daemon` "stops the daemon twice". It stops it in the function scope, but that is not enough, because if `with_daemon` would run after that in another function, that would start the daemon again. So we need the session scoped fixture to stop the daemon at the very end. Anyway, `stop_daemon` is a no-op if the daemon is not running. To make it more clear, I could add a `if not daemon_client.is_daemon_running` guard before the `stop_daemon` call in the `finally` clause.

Reply:
Ok!